tests/unit_tests/test_game_rating.py

```diff
@@ -984,10 +1047,292 @@ async def test_single_wrong_report_still_rated_correctly(game: Game, player_fact
+            1: {RatingType.GLOBAL: Rating(2000, 75)},
+            2: {RatingType.GLOBAL: Rating(1900, 75)},
+        },
+        expected_results=[
+            PersistedResults(
+                rating_type=RatingType.LADDER_1V1,
+                ratings={
+                    1: pytest.approx((1766, 429), abs=1),
+                    2: pytest.approx((1235, 429), abs=1),
+                },
+                outcomes=IGNORED
+            ),
+            PersistedResults(
+                rating_type=RatingType.GLOBAL,
+                ratings={},
+                outcomes=IGNORED
+            )
+        ]
+    )
+
+
+async def test_rating_adjustment_1v1_ladder_pros_global_newbies(
+    rating_service, player_factory
+):
+    IGNORED = object()
+
+    # Both players have high ladder rating but no global rating
+    await do_test_rating_adjustment(
+        rating_service,
+        player_factory,
+        ratings={
+            1: {RatingType.LADDER_1V1: Rating(2000, 75)},
+            2: {RatingType.LADDER_1V1: Rating(1900, 75)},
+        },
+        expected_results=[
+            PersistedResults(
+                rating_type=RatingType.LADDER_1V1,
+                ratings={
+                    1: pytest.approx((2011, 75), abs=1),
+                    2: pytest.approx((1889, 75), abs=1),
+                },
+                outcomes=IGNORED
+            ),
+            PersistedResults(
+                rating_type=RatingType.GLOBAL,
+                ratings={
+                    1: pytest.approx((2038, 348), abs=1),
+                    2: pytest.approx((1340, 419), abs=1),
+                },
+                outcomes=IGNORED
+            )
+        ]
+    )
+
+
+async def test_rating_adjustment_1v1_ladder_pros_global_mixed(
+    rating_service, player_factory
+):
+    IGNORED = object()
+
+    # Both players have high ladder rating, one has low global, the other has
+    # high global.
+    await do_test_rating_adjustment(
+        rating_service,
+        player_factory,
+        ratings={
+            1: {
+                RatingType.LADDER_1V1: Rating(2000, 75),
+                RatingType.GLOBAL: Rating(1000, 250),
+            },
+            2: {
+                RatingType.LADDER_1V1: Rating(1900, 75),
+                RatingType.GLOBAL: Rating(2100, 75),
+            }
+        },
+        expected_results=[
+            PersistedResults(
+                rating_type=RatingType.LADDER_1V1,
+                ratings={
+                    1: pytest.approx((2011, 75), abs=1),
+                    2: pytest.approx((1889, 75), abs=1),
+                },
+                outcomes=IGNORED
+            ),
+            PersistedResults(
+                rating_type=RatingType.GLOBAL,
+                ratings={
+                    1: pytest.approx((1373, 208), abs=1),
+                },
+                outcomes=IGNORED
+            )
+        ]
+    )
+
+
+async def test_rating_adjustment_1v1_ladder_pro_vs_global_pro(
```

Comment:
Not really sure what you want to demonstrate with this test...

Reply:
Mostly to show a situation where the global rating adjustment is going to be pretty huge. The adjustment will be rated as if a 0 rated player (1500, 500) was able to beat a 1600 rated player (1900, 75) in a 1v1. The 0 rated player's displayed rating will jump to 479 in a single game.

Reply:
This actually shows a bug. We want to have the player "play" with his global rating against the ladder rating of player 2 for the adjustment. So it should be the result as if two (1500, 500) players played against each other.
However the code looks like it should do exactly that, so I am not sure where the bug is.

Reply:
Oh I guess you're right. It's not bugged though because (1766, 429) is actually just what the winner of a (1500, 500) vs (1500, 500) game gets. I'll refactor that to a constant to make it more clear